File: MyLeetCodeJS/BubbleSort.js
const BubbleSort = (arr, n) => {
  let done = false;
  while (!done) {
    done = true;
    i = 0;
    while (i < n - 1) {
      if (arr[i] > arr[i + 1]) {
        //swapping the elements
        let c = arr[i + 1];
        arr[i + 1] = arr[i];
        arr[i] = c;
        done = false
      }
      i++;
    }
  }
  return arr

};


console.log(BubbleSort([5,4,3,2,1],5))

File: MyLeetCodePY/bubbleSort.py
def BubbleSort(arr, n):
    done = False
    while not done:
        done = True
        i = 0
        while i < n - 2:
            if arr[i] > arr[i + 2]:
                c = arr[i + 2]
                arr[i + 2] = arr[i]
                arr[i] = c
                done = False

            i += 1

    return arr


for i in range(int(input())):
    n = int(input())
    arr = [int(x) for x in input().strip().split()]
    print(BubbleSort(arr, n))
